sources/packages/backend/apps/queue-consumers/src/processors/schedulers/ier-Integrations/ier-integration.scheduler.ts

```diff
@@ -0,0 +1,75 @@
+import { InjectQueue, OnQueueActive, Process, Processor } from "@nestjs/bull";
+import { IER12FileService } from "@sims/integrations/institution-integration/ier-integration";
+import {
+  IER_SCHEDULER_JOB_ID,
+  QUEUE_RETRY_DEFAULT_CONFIG,
+} from "@sims/services/constants";
+import { QueueNames } from "@sims/services/queue";
+import { PST_TIMEZONE } from "@sims/utilities";
+import { ConfigService } from "@sims/utilities/config";
+import { InjectLogger, LoggerService } from "@sims/utilities/logger";
+import Bull, { CronRepeatOptions, Job, Queue } from "bull";
+import { BaseScheduler } from "../base-scheduler";
+import {
+  GeneratedDateQueueInDTO,
+  IER12ResultQueueOutDTO,
+} from "./models/ier.model";
+
+@Processor(QueueNames.IERIntegration)
+export class IERIntegrationScheduler extends BaseScheduler<GeneratedDateQueueInDTO> {
+  cronOptions: Bull.JobOptions = undefined;
+
+  constructor(
+    @InjectQueue(QueueNames.IERIntegration)
+    readonly schedulerQueue: Queue<GeneratedDateQueueInDTO>,
+    private readonly ierRequest: IER12FileService,
+    config: ConfigService,
+  ) {
+    super();
+    // cron options.
+    this.cronOptions = {
```

Comment:
I believe that we need to review how to handle the `this.cronOptions` and the `schedulerQueue`.

For the queue, maybe the injected queue can be passed as a parameter to the constructor of the base class using the `super(...)` and the base class can have it as a protected property.

For the `cronOptions` maybe it must be handled inside the `initializeScheduler`.
This can be impacted by the upcoming changes to have the queues configured from DB. Maybe we can try to settle this as simply as possible e work on changes in the upcoming ticket.